src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -704,12 +699,7 @@ void C2_MacroAssembler::fast_lock(Register objReg, Register boxReg, Register tmp
 
   bind(COUNT);
   // Count monitors in fast path
-#ifndef _LP64
-  get_thread(tmpReg);
-  incrementl(Address(tmpReg, JavaThread::held_monitor_count_offset()));
-#else // _LP64
-  incrementq(Address(r15_thread, JavaThread::held_monitor_count_offset()));
-#endif
+  increment(Address(thread, JavaThread::held_monitor_count_offset()));
```

Comment:
Okay you updated this logic to use the `thread` register that's passed in, but
there are two other uses of `r15_thread` in `C2_MacroAssembler::fast_lock()`
on L679 and L686 that could also be switched to `thread`.